### PR TITLE
feat: multi mode chart viewer

### PR DESCRIPTION
**Pull request by boojack**
For customizing our timeline chart, I rewrite the chart-xkcd's XY chart with TypeScript.

### CODE REVIEW

**Comment by boojack** [comment]
PTAL

File: src/components/Charts/StarXYChart.vue [hunk 1]
```diff
@@ -8,56 +8,45 @@
 </template>
 
 <script lang="ts">
-import ChartXkcd from "chart.xkcd";
 import { defineComponent, onMounted, onUpdated, ref } from "vue";
+import XYChart, { XYChartData } from "../../packages/xy-chart";
```
Comment (tianzhou) [inline src/components/Charts/StarXYChart.vue:12]: Let's add a comment here explaining the reason we have to develop our own XYChart

**Review by tianzhou — CHANGES_REQUESTED** [review]
I also notice a separate bug when testing this https://star-history.com/#bytebase/star-history&bytebase/bytebase&oceanbase/oceanbase

The yellow line doesn't start from 0

CleanShot 2022-01-05 at 11 41 44

**Comment by boojack** [comment]
PTAL

File: src/components/Charts/StarXYChart.vue (hunk 1 above)
Comment (tianzhou) [reply src/components/Charts/StarXYChart.vue:14]: This comment doesn't clearly mention *WHY*.  Essentially what's the limitation of the existing xkcd chart, which forces us to rewrite it.

**Review by tianzhou — CHANGES_REQUESTED** [review]
Please refine the comment.

**Comment by boojack** [comment]
PTAL

File: src/components/Charts/StarXYChart.vue [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+import { defineComponent, onMounted, onUpdated, ref } from "vue";
+// For customizing multi mode chart: base on create date or timeline, we have rewrited the chart.xkcd's XY chart with TypeScript.
+// Here are some reasons about this motivation.
+//
+// The shortcomings of the old chart.xkcd (project link: https://github.com/timqian/chart.xkcd):
+// 1. The X axis label string could be only formatted date or raw text;
+// 2. It's difficult/not possible to debug and customize;
+//
+// The advantages of the new one:
+// 1. Writing in TypeScript originally;
```
Comment (tianzhou) [inline src/components/Charts/StarXYChart.vue:20]: 1. Writing in native TypeScript;
2. Easy to debug chart internal;
3. Totally customizable.

**Review by tianzhou — APPROVED** [review]
LGTM

**Comment by tianzhou** [comment]
This is a nift change!

**Pull request merged**